### PR TITLE
[extractor/zoom] Update extraction logic

### LINKED ISSUE #6677: [zoom.us] Fails with No video formats found!
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that a **supported** site is broken
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

anywhere

### Provide a description that is worded well enough to be understood

Test link:
https://ffgolf.zoom.us/rec/play/dJy6QRkm1fsfZCx2h8cHNJh7N1P0NP2alj-0XDL3c2cPdtAO8KK7lFdcmRod1fNxqIdzbPK3fLnMoi_w.HrmfpuxLVWa26JCt

No geo restriction or password needed.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
./yt-dlp -vU -F "https://ffgolf.zoom.us/rec/play/dJy6QRkm1fsfZCx2h8cHNJh7N1P0NP2alj-0XDL3c2cPdtAO8KK7lFdcmRod1fNxqIdzbPK3fLnMoi_w.HrmfpuxLVWa26JCt"
[debug] Command-line config: ['-vU', '-F', 'https://ffgolf.zoom.us/rec/play/dJy6QRkm1fsfZCx2h8cHNJh7N1P0NP2alj-0XDL3c2cPdtAO8KK7lFdcmRod1fNxqIdzbPK3fLnMoi_w.HrmfpuxLVWa26JCt']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.03 [934496428] (zip)
[debug] Python 3.11.2 (CPython arm64 64bit) - macOS-13.2.1-arm64-arm-64bit (OpenSSL 1.1.1t  7 Feb 2023)
[debug] exe versions: phantomjs 2.1.1
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Extractor Plugins: SamplePluginIE
[debug] Post-Processor Plugins: SamplePluginPP
[debug] Loaded 1850 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.03
Current Build Hash: c3b48730357dee54f39c679566261eb2bbfd96c32a1d25513efd04b76a9ecf42
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp/releases/latest/download/_update_spec
Updating to stable@2023.03.04 ...
[debug] Downloading yt-dlp from https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp
[debug] Downloading SHA2-256SUMS from https://github.com/yt-dlp/yt-dlp/releases/latest/download/SHA2-256SUMS
Updated yt-dlp to stable@2023.03.04
[debug] Restarting: /opt/homebrew/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/Resources/Python.app/Contents/MacOS/Python ./yt-dlp -vU -F https://ffgolf.zoom.us/rec/play/dJy6QRkm1fsfZCx2h8cHNJh7N1P0NP2alj-0XDL3c2cPdtAO8KK7lFdcmRod1fNxqIdzbPK3fLnMoi_w.HrmfpuxLVWa26JCt
[debug] Command-line config: ['-vU', '-F', 'https://ffgolf.zoom.us/rec/play/dJy6QRkm1fsfZCx2h8cHNJh7N1P0NP2alj-0XDL3c2cPdtAO8KK7lFdcmRod1fNxqIdzbPK3fLnMoi_w.HrmfpuxLVWa26JCt']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (zip)
[debug] Python 3.11.2 (CPython arm64 64bit) - macOS-13.2.1-arm64-arm-64bit (OpenSSL 1.1.1t  7 Feb 2023)
[debug] exe versions: phantomjs 2.1.1
[debug] Optional libraries: no_Cryptodome-None, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Extractor Plugins: SamplePluginIE
[debug] Post-Processor Plugins: SamplePluginPP
[debug] Loaded 1787 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
Current Build Hash: 91cad9f121c1f6f0a81b747415c46ecba0ff331ed38cc6433040b4ac7b6e15ca
yt-dlp is up to date (stable@2023.03.04)
[zoom] Extracting URL: https://ffgolf.zoom.us/rec/play/dJy6QRkm1fsfZCx2h8cHNJh7N1P0NP2alj-0XDL3c2cPdtAO8KK7lFdcmRod1fNxqIdzbPK3fLnMoi_w.HrmfpuxLVWa26JCt
[zoom] dJy6QRkm1fsfZCx2h8cHNJh7N1P0NP2alj-0XDL3c2cPdtAO8KK7lFdcmRod1fNxqIdzbPK3fLnMoi_w.HrmfpuxLVWa26JCt: Downloading webpage
WARNING: Extractor failed to obtain "title". Creating a generic title instead
ERROR: [zoom] dJy6QRkm1fsfZCx2h8cHNJh7N1P0NP2alj-0XDL3c2cPdtAO8KK7lFdcmRod1fNxqIdzbPK3fLnMoi_w.HrmfpuxLVWa26JCt: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
Traceback (most recent call last):
  File "./yt-dlp/yt_dlp/YoutubeDL.py", line 1518, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/YoutubeDL.py", line 1615, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/YoutubeDL.py", line 1674, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/YoutubeDL.py", line 2615, in process_video_result
    self.raise_no_formats(info_dict)
  File "./yt-dlp/yt_dlp/YoutubeDL.py", line 1046, in raise_no_formats
    raise ExtractorError(msg, video_id=info['id'], ie=info['extractor'],
yt_dlp.utils.ExtractorError: [zoom] dJy6QRkm1fsfZCx2h8cHNJh7N1P0NP2alj-0XDL3c2cPdtAO8KK7lFdcmRod1fNxqIdzbPK3fLnMoi_w.HrmfpuxLVWa26JCt: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```

### PR DESCRIPTION
**Pull request by shreyasminocha**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

The zoom extractor broke recently due to upstream changes. This is an attempt to fix that.

Fixes #6677

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by shreyasminocha** [comment]
Downloading simple non-password-protected works, but I haven't tested these yet:

- subtitle extraction
- password-protected videos
- downloading the screen share video

File: yt_dlp/extractor/zoom.py [hunk 1]
```diff
@@ -53,42 +53,48 @@ def _real_extract(self, url):
...
             formats.append({
                 'format_note': 'Screen share stream',
-                'url': str_or_none(data.get('shareMp4Url')),
-                'width': int_or_none(data.get('shareResolvtionsWidth')),
-                'height': int_or_none(data.get('shareResolvtionsHeight')),
-                'format_id': str_or_none(data.get('shareVideoId')),
+                'url': str_or_none(result.get('shareMp4Url')),
+                'width': int_or_none(result.get('shareResolvtions')[0]),
+                'height': int_or_none(result.get('shareResolvtions')[1]),
+                'format_id': str_or_none(result.get('shareVideo', {}).get('id')),
```
Comment (pukkandan) [inline yt_dlp/extractor/zoom.py:90]: `a.get(b)[1]` etc are not safe. use traverse_obj

File: yt_dlp/extractor/zoom.py [hunk 2]
```diff
@@ -53,42 +53,48 @@ def _real_extract(self, url):
             r'(?s)window\.__data__\s*=\s*({.+?});',
             webpage, 'data'), play_id, js_to_json)
 
+        info = self._download_json(
+            base_url + 'nws/recording/1.0/play/info/' + data['fileId'], play_id
+        )
+
+        result = info.get('result', {})
```
Comment (pukkandan) [inline yt_dlp/extractor/zoom.py:60]:
```suggestion
        data = self._download_json(
            f'{base_url}nws/recording/1.0/play/info/{data["fileId"]}', play_id)['result']
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/zoom.py [hunk 3]
```diff
@@ -67,28 +71,29 @@ def _real_extract(self, url):
...
-                'width': int_or_none(data.get('viewResolvtionsWidth')),
-                'height': int_or_none(data.get('viewResolvtionsHeight')),
-                'format_id': str_or_none(data.get('recordingId')),
+                'width': int_or_none(traverse_obj(data, ('viewResolvtions', 0))),
+                'height': int_or_none(traverse_obj(data, ('viewResolvtions', 1))),
+                'format_id': str_or_none(traverse_obj(data, ('recording', 'id'))),
                 'ext': 'mp4',
-                'filesize_approx': parse_filesize(data.get('fileSize')),
+                'filesize_approx': parse_filesize(
+                    traverse_obj(data, ('recording', 'fileSizeInMB'), expected_type=str_or_none)),
```
Comment (pukkandan) [inline yt_dlp/extractor/zoom.py:79]:
```suggestion
                'filesize_approx': traverse_obj(data, (
                    'recording', 'fileSizeInMB', {str_or_none}, {parse_filesize})),
```

File: yt_dlp/extractor/zoom.py [hunk 4]
```diff
@@ -67,28 +71,29 @@ def _real_extract(self, url):
...
+                'height': int_or_none(traverse_obj(data, ('shareResolvtions', 1))),
+                'format_id': str_or_none(traverse_obj(data, ('shareVideo', 'id'))),
                 'ext': 'mp4',
                 'preference': -1
             })
 
         return {
             'id': play_id,
-            'title': data.get('topic'),
+            'title': traverse_obj(data, ('meet', 'topic'), expected_type=str_or_none),
```
Comment (pukkandan) [inline yt_dlp/extractor/zoom.py:96]:
```suggestion
            'title': str_or_none(traverse_obj(data, ('meet', 'topic'))),
```

**Review by pukkandan — APPROVED** [review]

File: yt_dlp/extractor/zoom.py [hunk 5]
```diff
@@ -67,28 +71,29 @@ def _real_extract(self, url):
...
-                'width': int_or_none(data.get('viewResolvtionsWidth')),
-                'height': int_or_none(data.get('viewResolvtionsHeight')),
-                'format_id': str_or_none(data.get('recordingId')),
+                'width': int_or_none(traverse_obj(data, ('viewResolvtions', 0))),
+                'height': int_or_none(traverse_obj(data, ('viewResolvtions', 1))),
+                'format_id': str_or_none(traverse_obj(data, ('recording', 'id'))),
                 'ext': 'mp4',
-                'filesize_approx': parse_filesize(data.get('fileSize')),
+                'filesize_approx': traverse_obj(data, (
+                    'recording', 'fileSizeInMB', {str_or_none}, {parse_filesize})),
```
Comment (pukkandan) [inline yt_dlp/extractor/zoom.py:79]:
```suggestion
                'filesize_approx': parse_filesize(str_or_none(traverse_obj(data, ('recording', 'fileSizeInMB')))),
```

**Pull request merged**